### ISSUE TITLE
How to improve the training of controlnet

### ISSUE THREAD

**Issue by garychan22** [huggingface/diffusers#3009]
Hi, thanks for the implementation of contorlnet in diffusers [https://github.com/huggingface/diffusers/blob/main/examples/controlnet/train_controlnet.py](url). I have tried the example with batch size 4 on A100 40G but I cannot produce good samples. The following is one of the generated sample

`cyan circle with brown floral background`
![output_cond2_v2_scale5](https://user-images.githubusercontent.com/108175311/230604926-e8cacd5d-9383-4feb-8bfa-bcffdb1f4c11.png)

How can I improve the training of controlnet to produce good results? Thanks a lot!

**Comment by patrickvonplaten** [comment]
cc @yiyixuxu

**Comment by yiyixuxu** [comment]
Hi @garychan22 

you could try different learning schedules, bigger batch sizes, and gradient accumulation 

We use this dataset as a sanity check and a super easy example to demo how the training script works - I don't think it's the best dataset to understand the controlnet training dynamic
